base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -673,6 +694,143 @@ where
         Ok(tx_id)
     }
 
+    /// Sends a one side payment transaction to a recipient
+    /// # Arguments
+    /// 'dest_pubkey': The Comms pubkey of the recipient node
+    /// 'amount': The amount of Tari to send to the recipient
+    /// 'fee_per_gram': The amount of fee per transaction gram to be included in transaction
+    pub async fn send_one_sided_transaction(
+        &mut self,
+        dest_pubkey: CommsPublicKey,
+        amount: MicroTari,
+        fee_per_gram: MicroTari,
+        message: String,
+        transaction_broadcast_join_handles: &mut FuturesUnordered<
+            JoinHandle<Result<u64, TransactionServiceProtocolError>>,
+        >,
+    ) -> Result<TxId, TransactionServiceError>
+    {
+        if self.node_identity.public_key() == &dest_pubkey {
+            warn!(target: LOG_TARGET, "One-sided spend-to-self transactions not supported");
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "One-sided spend-to-self transactions not supported".to_string(),
+            ));
+        }
+
+        // Prepare sender part of the transaction
+
+        let mut stp = self
+            .output_manager_service
+            .prepare_transaction_to_send(
+                amount,
+                fee_per_gram,
+                None,
+                message.clone(),
+                script!(PushPubKey(Box::new(dest_pubkey.clone()))),
+            )
+            .await?;
+        let tx_id = stp.get_tx_id()?;
+        if !stp.is_single_round_message_ready() {
```

Comment:
Under what conditions can this if statement ever be true?

Reply:
Because we do not use the series of transaction protocols over the wire directly here, certain checks are needed to make sure the sender transaction protocol has the correct state. 

This test is specifically inserted to let a future maintainer know that the sender transaction protocol should be in a specific state. This is a cheap operation and worth much more than a comment. If something is changed in the transaction protocols that influence the state, this test must still pass.

Reply:
Is there another thread that can access this data somehow? Surely the object is only available to be edited by this code in the method? These extra checks take up space and clutter the actual code that is necessary

Reply:
But, you are correct, the methods are deterministic, so if it advances beyond the error check, the test is not needed.